### ISSUE TITLE
subtype dict unpack

### ISSUE THREAD

**Issue by Akuli** [python/mypy#10171]
**Bug Report**

**To Reproduce**
```
from typing import Dict
a: Dict[int, str] = {}
b: Dict[object, str] = {object(): 'a', 'lol': 'b', **a}
```
**Expected Behavior**
no error

**Actual Behavior**
confusing error message `Argument 1 to "update" of "dict" has incompatible type "Dict[int, str]"; expected "Mapping[object, str]"` even though update method was not used

**Your Environment**

- Mypy version used: 0.812
- Mypy command-line flags: none
- Mypy configuration options from `mypy.ini` (and other config files): none
- Python version used: 3.7.3

**Comment by Parth-Shete** [comment]
You can use 'char' data type instead of 'str' as you are using only a single alphabet.
And you also have to import the 'object' along with the 'Dict'.

If you still didn't get it then refer this page as provided by the link...,
[https://stackoverflow.com/questions/43910979/mypy-error-incompatible-types-in-assignment#:~:text=1%20Answer&text=The%20problem%20is%20that%20mypy,%2C%20mypy%20(rightfully)%20complains.&text=This%20isn't%20fully%20typesafe,some%20checks%20on%20your%20dict.](url)

**Comment by Akuli** [comment]
@Parth-Shete  Your comment doesn't make any sense and is not relevant to my issue in any way. Python does not have a `char` data type. It has a built-in named `object` which does not have to be imported, because it is a built-in.

**Comment by Parth-Shete** [comment]
You can use the link which provides the required information about this issue from stackoverflow.

**Comment by Akuli** [comment]
No, that is not the same issue. What you're saying still doesn't make any sense.

**Comment by Parth-Shete** [comment]
Oh okay my bad.
